dask/dataframe/utils.py

```diff
@@ -444,6 +466,26 @@ def _nonempty_series(s, idx=None):
             cats = None
         data = pd.Categorical(data, categories=cats,
                               ordered=s.cat.ordered)
+    elif is_integer_na_dtype(dtype):
+        data = pd.array([1, None], dtype=dtype)
+    elif is_period_dtype(dtype):
+        # pandas 0.24.0+ should infer this to be Series[Period[freq]]
+        freq = dtype.freq
+        data = [pd.Period('2000', freq), pd.Period('2001', freq)]
+    elif is_sparse(dtype):
+        # TODO: pandas <0.24
```

Comment:
Grr this is out of date. But I'm planning to rewrite all this soon, so just going to leave it for now.